src/components/Result.js

```diff
@@ -11,7 +11,12 @@ function Result({ result, dispatch }) {
     );
   }
 
-  if (!result.expression) {
+  if (
+    !result.expression ||
+    !result.elements ||
+    !Array.isArray(result.elements) ||
+    result.elements.length === 0
```

Comment:
The array check makes the `undefined` check redundant.

```suggestion
    !result.expression ||
    !Array.isArray(result.elements) ||
    result.elements.length === 0
```